src/main/java/fi/aalto/cs/apluscourses/model/Tutorial.java

```diff
@@ -45,4 +75,97 @@ public static Tutorial fromJsonObject(@NotNull JSONObject jsonObject) {
   public void onComplete() {
     this.tutorialCompleted.trigger();
   }
+
+  /**
+   * Downloads the module deps for a tutorial, with the module name starting with "Ideact_".
+   */
+  public void downloadDependencies(@NotNull Course course,
+                                   @NotNull Project project,
+                                   @NotNull TaskNotifier taskNotifier) {
+    taskNotifier.notifyDownloadingDeps(false);
+
+    var modules = getModules(course);
+
+    var componentInstallerFactory = new ComponentInstallerImpl.FactoryImpl<>(new SimpleAsyncTaskManager());
+    var dialogs = new ReinstallDialogs(project);
```

Comment:
These should be given as parameters. Actually, the functionality of this whole method would better be handled in the caller (the action).